File: 102.py
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if (not root):
            return []

        queue = [root]
        arr = []

        while (len(queue)):
            level = []

            for i in range(0, len(queue)):
                shiftedNode = queue.pop(0)
                level.append(shiftedNode.val)

                if (shiftedNode.left):
                    queue.append(shiftedNode.left)

                if (shiftedNode.right):
                    queue.append(shiftedNode.right)

            arr.append(level)

        return arr

File: Leetcode/Binary Trees/102.js
// 102. Binary Tree Level Order Traversal

const levelOrder = (root) => {
  if (!root) {
    return [];
  }

  const queue = [root];
  const arr = [];

  while (queue.length) {
    const lengthOfQueue = queue.length;
    const temp = [];

    for (let i = 0; i < lengthOfQueue; i++) {
      const removedNode = queue.shift();

      temp.push(removedNode.val);

      if (removedNode.left) {
        queue.push(removedNode.left);
      }

      if (removedNode.right) {
        queue.push(removedNode.right);
      }
    }

    arr.push(temp);
  }

  return arr;
};

// Time Complexity: O(n)
// Space Complexity: O(n)
